src/v/cloud_storage/cache_service.cc

```diff
@@ -720,12 +724,26 @@ cache::trim_exhaustive(uint64_t size_to_delete, size_t objects_to_delete) {
         } catch (const ss::gate_closed_exception&) {
             // We are shutting down, stop iterating and propagate
             throw;
+        } catch (const std::filesystem::filesystem_error& e) {
+            if (likely(file_stat.path.ends_with(tmp_extension))) {
+                // In exhaustive scan we might hit a .part file and get ENOENT,
+                // this is expected behavior occasionally.
+                result.trim_missed_tmp_files = true;
```

Comment:
I wonder if it makes sense to catch ENOENT specifically and count the number of files that were looked for, but removed. From a trimming perspective, regardless of tmp file or not, maybe that should be what we use to determine if it's an error or not (at least it feels more intuitive than checking for tmp files)

I don't feel too strongly about it, but worth considering, i think

Reply:
I think this makes sense, I will create a follow up issue for doing this.

Reply:
https://github.com/redpanda-data/redpanda/issues/13908